src/Cell.cpp

```diff
@@ -1434,6 +1438,7 @@ void Cell::ringify(std::vector<Cell*>& subcells, double max_radius) {
       Cell* ring = clone();
       ring->setNumSectors(0);
       ring->setNumRings(0);
+      ring->removeSurface(zcylinder1);
```

Comment:
Not solve this issue completely. It is hard to remove the abundant outer box surfaces for outer most cells ( moderator usually). However, this could be improved more.

Reply:
It is the problem when a point is on the surface. In this case, the evaluation value would be so tiny that it could be either minus or positive. Will be fixed in other PR.

Reply:
This removeSurfaces doesn't actually work. It's calling getSurfaces which returns a const map, so you can't delete from it. 

No need to make a PR to fix it, all of this is gone with Regions